src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -19,96 +23,119 @@ namespace Microsoft.Health.SqlServer.Registration
+            services.TryAddSingleton(p => p.GetServices<IHostedService>().First(x => x is SchemaInitializer) as SchemaInitializer);
+            services.TryAddSingleton(p => p.GetRequiredService<IScriptProvider>() as ScriptProvider<TSchemaVersionEnum>);
+            services.TryAddSingleton(p => p.GetRequiredService<IBaseScriptProvider>() as BaseScriptProvider);
+            services.TryAddSingleton(p => p.GetRequiredService<ISchemaManagerDataStore>() as SchemaManagerDataStore);
+            services.TryAddSingleton(p => p.GetRequiredService<IPollyRetryLoggerFactory>() as PollyRetryLoggerFactory);
+            services.TryAddSingleton(p => p.GetRequiredService<SqlCommandWrapperFactory>() as RetrySqlCommandWrapperFactory);
+            services.TryAddSingleton(p => p.GetRequiredService<ISqlServerTransientFaultRetryPolicyFactory>() as SqlServerTransientFaultRetryPolicyFactory);
+
+            return services;
+        }
+
+        /// <summary>
+        ///  Adds a collection of services for connecting to SQL Server to the specified <see cref="IServiceCollection"/>.
+        /// </summary>
+        /// <param name="services">The <see cref="IServiceCollection"/> to be updated.</param>
+        /// <returns>The <paramref name="services"/> for additional method invocations.</returns>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="services"/> is <see langword="null"/>.
+        /// </exception>
+        public static IServiceCollection AddSqlServerConnection(this IServiceCollection services)
+        {
+            EnsureArg.IsNotNull(services, nameof(services));
+
+            services.AddOptions();
+            services.TryAddSingleton<ISqlConnectionStringProvider, DefaultSqlConnectionStringProvider>();
+            services.TryAddSingleton<ISqlConnectionFactory>(
+                p =>
+                {
+                    SqlServerDataStoreConfiguration config = p.GetRequiredService<IOptions<SqlServerDataStoreConfiguration>>().Value;
+                    ISqlConnectionStringProvider sqlConnectionStringProvider = p.GetRequiredService<ISqlConnectionStringProvider>();
+                    return config.AuthenticationType == SqlServerAuthenticationType.ManagedIdentity
+                        ? new ManagedIdentitySqlConnectionFactory(sqlConnectionStringProvider, p.GetRequiredService<IAccessTokenHandler>())
+                        : new DefaultSqlConnectionFactory(sqlConnectionStringProvider);
+                });
+
+            // The following are only used in case of managed identity
+            services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>();
+            services.AddSingleton<AzureServiceTokenProvider>();
+
+            // Services to facilitate SQL connections
+            // TODO: Does SqlTransactionHandler need to be registered directly? Should usage change to ITransactionHandler?
```

Comment:
The services in this project take a dependency on `SqlTransactionHandler` directly instead of `ITransactionHandler`. I think this should be changed